src/protocols/core/Compositor.cpp

```diff
@@ -84,12 +84,6 @@ CWLSurfaceResource::CWLSurfaceResource(SP<CWlSurface> resource_) : resource(reso
             pending.texture    = res && res->buffer ? res->buffer->texture : nullptr;
             pending.bufferSize = res && res->buffer ? res->buffer->size : Vector2D{};
         }
-
-        Vector2D oldBufSize = current.buffer ? current.bufferSize : Vector2D{};
-        Vector2D newBufSize = pending.buffer ? pending.bufferSize : Vector2D{};
-
-        if (oldBufSize != newBufSize || current.buffer != pending.buffer)
```

Comment:
the size check has to be enforced though

Reply:
I don't think the buffer needs updated, but yeah we might need this for the renderer to damage the monitor for the resize. I'll add it back in.

Reply:
In theory I think the client should damage the entire buffer when resizing, but it can't hurt to also damage it here.

Reply:
we should just damage the buffer on resize thats about it. IIRC it was somewhere in the spec? might be wrong tho

Reply:
I don't think that's in the spec, but it can't hurt. I added back the size check (slightly improved imo) and also damaged the buffer on scale or transform changes.